commit d07464e5b84d91b8f81a4f0fd4912e1bd2cd8dc2
Author: Umaro <umaro@narshe.net>
Date:   Wed Jun 22 16:41:00 2011 +0200

    USB: gadget: g_mass_storage: static data instead of dynamic allocation
    
        This patch changes msg_do_config() function so that it uses
        a static object for a fsg_common structure instead of dynamically
        allocated.  This is a micro-optimisation.
    
        Signed-off-by: Michal Nazarewicz <m.nazarewicz@samsung.com>
        Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
        Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/drivers/usb/gadget/mass_storage.c b/drivers/usb/gadget/mass_storage.c
index 705cc1f..5f03df8 100644
--- a/drivers/usb/gadget/mass_storage.c
+++ b/drivers/usb/gadget/mass_storage.c
@@ -143,7 +143,9 @@ static int msg_thread_exits(struct fsg_common *common)
 
 static int __init msg_do_config(struct usb_configuration *c)
 {
-	struct fsg_common *common;
+	static struct fsg_common common;
+
+	struct fsg_common *retp;
 	struct fsg_config config;
 	int ret;
 
@@ -154,12 +156,12 @@ static int __init msg_do_config(struct usb_configuration *c)
 
 	fsg_config_from_params(&config, &mod_data);
 	config.thread_exits = msg_thread_exits;
-	common = fsg_common_init(0, c->cdev, &config);
-	if (IS_ERR(common))
-		return PTR_ERR(common);
+	retp = fsg_common_init(&common, c->cdev, &config);
+	if (IS_ERR(retp))
+		return PTR_ERR(retp);
 
-	ret = fsg_add(c->cdev, c, common);
-	fsg_common_put(common);
+	ret = fsg_add(c->cdev, c, &common);
+	fsg_common_put(&common);
 	return ret;
 }
 
